src/orbit_determination/tooshortarc.jl

```diff
@@ -0,0 +1,549 @@
+- `v_α::T`: right ascension velocity.
+- `v_δ::T`: declination velocity.
+- `ρ_unit, ρ_α, ρ_δ::Vector{T}`: topocentric unit vector and its partials.
+- `q::Vector{T}`: heliocentric position of observer.
+- `coeffs::Vector{T}`: polynomial coefficients.
+- `ρ_domain::Vector{T}`: range domain.
+- `v_ρ_domain::Vector{T}`: range-rate domain.
+- `Fs::Matrix{T}`: boundary points.
+- `observatory::ObservatoryMPC{T}`: observing station.
+
+!!! reference
+    See Chapter 8 of https://doi.org/10.1017/CBO9781139175371.
+"""
+@auto_hash_equals struct AdmissibleRegion{T <: AbstractFloat}
+    date::DateTime
+    α::T
+    δ::T
+    v_α::T
+    v_δ::T
+    ρ_unit::Vector{T}
+    ρ_α::Vector{T}
+    ρ_δ::Vector{T}
+    q::Vector{T}
+    coeffs::Vector{T}
+    ρ_domain::Vector{T}
+    v_ρ_domain::Vector{T}
+    Fs::Matrix{T}
+    observatory::ObservatoryMPC{T}
+end
+
+# Definition of zero AdmissibleRegion{T}
+function zero(::Type{AdmissibleRegion{T}}) where {T <: AbstractFloat}
+    return AdmissibleRegion{T}(
+        DateTime(2000), zero(T), zero(T), zero(T), zero(T), Vector{T}(undef, 0),
+        Vector{T}(undef, 0), Vector{T}(undef, 0), Vector{T}(undef, 0), Vector{T}(undef, 0),
+        Vector{T}(undef, 0), Vector{T}(undef, 0), Matrix{T}(undef, 0, 0), unknownobs()
+    )
+end
+
+iszero(x::AdmissibleRegion{T}) where {T <: AbstractFloat} = x == zero(AdmissibleRegion{T})
+
+# Outer constructor
+function AdmissibleRegion(tracklet::Tracklet{T}, params::NEOParameters{T}) where {T <: AbstractFloat}
+    # Unfold
+    obs, t_datetime, α, δ = observatory(tracklet), date(tracklet), ra(tracklet), dec(tracklet)
+    v_α, v_δ, h = vra(tracklet), vdec(tracklet), mag(tracklet)
+    # Topocentric unit vector and partials
+    ρ, ρ_α, ρ_δ = topounitpdv(α, δ)
+    # Time of observation [days since J2000]
+    t_days = datetime2days(t_datetime)
+    # Time of observation [et seconds]
+    t_et = datetime2et(t_datetime)
+    # Heliocentric position of the observer
+    q = params.eph_ea(t_days) + kmsec2auday(obsposvelECI(obs, t_et)) - params.eph_su(t_days)
+    # Admissible region coefficients
+    coeffs = admsreg_coeffs(α, δ, v_α, v_δ, ρ, ρ_α, ρ_δ, q)
+    # Tiny object boundary
+    H_max = 32.0                # Maximum allowed absolute magnitude
+    if isnan(h)
+        ρ_min = R_SI
+    else
+        ρ_min = 10^((h - H_max)/5)
+    end
+    # Maximum range (heliocentric constraint)
+    ρ_max = max_range(coeffs, ρ_min)
+    iszero(ρ_max) && return zero(AdmissibleRegion{Float64})
+    # Range domain
+    ρ_domain = [ρ_min, ρ_max]
+    # Range rate domain
+    v_ρ_min, v_ρ_max = range_rate(coeffs, ρ_min)[1:2]
+    v_ρ_domain = [v_ρ_min, v_ρ_max]
+    # Range rate symmetry level
+    v_ρ_mid = range_rate(coeffs, ρ_max)[1]
+    # Boundary points
+    Fs = Matrix{Float64}(undef, 3, 2)
+    Fs[1, :] .= [ρ_min, v_ρ_min]
+    Fs[2, :] .= [ρ_min, v_ρ_max]
+    Fs[3, :] .= [ρ_max, v_ρ_mid]
+    
+    return AdmissibleRegion{Float64}(t_datetime, α, δ, v_α, v_δ, ρ, ρ_α, ρ_δ, q,
```

Comment:
Since the method has the type parameter `T<:AbstractFloat` I suggest to also use it for the returned value
```suggestion
    return AdmissibleRegion{T}(t_datetime, α, δ, v_α, v_δ, ρ, ρ_α, ρ_δ, q,
```

Reply:
Done.